algorithms/a3c/main.py

```diff
@@ -40,80 +54,276 @@
+        elif args.evaluate == 1:
+            args.use_train_instructions = 1
+            args.num_processes = 0
+        elif args.evaluate == 2:
+            args.use_train_instructions = 0
+            args.num_processes = 0
+        else:
+            assert False, "Invalid evaluation type"
+
+        env = GroundingEnv(args)
+        env.game_init()
+        args.resolution = (args.frame_width, args.frame_height)
     else:
-        args.config_dict = {'max_episode_length': args.max_episode_length}
-        env = AI2ThorEnv(config_dict=args.config_dict)
-        args.frame_dim = env.config['resolution'][-1]
-    shared_model = ActorCritic(env.observation_space.shape[0], env.action_space.n, args.frame_dim)
-    shared_model.share_memory()
+        # if you resume a checkpoint and if args.resume_latest_config == True, changes to the file
+        # will be overwritten by the latest_config.json file. Set to False if you want to change
+        # config settings in the config file after resuming and in the middle of training
+        args.last_config_resume_path = os.path.join(args.experiment_path, 'latest_config.json')
+        args.config_dict = {}
+        if args.resume_latest_config:
+            print('args.resume_latest_config is set on')
+            # read last_config.json if it exists, else create it below
+            if os.path.exists(args.last_config_resume_path):
+                print('Folder for this args.experiment_id "{}" and latest_config file existed'
+                      .format(args.experiment_id))
+                with open(args.last_config_resume_path, 'r') as f:
+                    print('Loading config_dict from file: {}'.format(args.last_config_resume_path))
+                    print('Therefore changes to args.config_file_name won\'t affect the env.\n'
+                          'Set to args.resume_latest_config to 0 if you don\'t want this')
+                    args.config_dict = json.load(f)
+        else:
+            print('Loading most params from args.config-file-name')
 
-    env.close()  # above env initialisation was only to find certain params needed
+        # random actions on reset to encourage robustness
+        args.config_dict['num_random_actions_at_init'] = args.num_random_actions_at_init
+        args.config_dict['max_episode_length'] = args.max_episode_length
+        # use given config file to start config and then allow config_dict to overwrite values
+        config_file_dir_path = os.path.abspath(os.path.join(__file__, '../../..', 'gym_ai2thor',
+                                                            'config_files'))
+        args.config_file_path = os.path.join(config_file_dir_path, args.config_file_name)
+        env = AI2ThorEnv(config_file=args.config_file_path, config_dict=args.config_dict,
+                         seed=args.seed)  # todo decide
+        args.resolution = (env.config['resolution'][0], env.config['resolution'][1])
 
+    # Create shared model
```

Comment:
Add an explanation like `A3C uses a shared model that is updated with the results of rollouts in parallel blabla.` Put some context to what "shared model" means